stdlib/public/core/StringUTF8.swift

```diff
@@ -48,8 +48,10 @@ extension _StringCore {
         src: UnsafeMutablePointer(startASCII + i),
         size: numericCast(utf16Count))
 
-      return (i + utf16Count, result)
+      // Convert the _UTF8Chunk into host endianness.
+      return (i + utf16Count, _UTF8Chunk(littleEndian: result))
```

Comment:
I think, we have another endianness problem in `_StringCore._nthContiguous`.
Sorry, I cannot test myself.